From 93d28fccac989d9947737dd76a026b16ea1c71b1 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Tue, 14 Apr 2020 10:53:38 +0200
Subject: [PATCH] bootscript: Disable sd boot

Drop SD-card boot as it will not be used anymore.

Still allowing to boot BalenaOS with non-balena
u-boot, to allow for writing the u-boot from the
hostOS

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 .../boundary/bootscripts/bootscript-yocto.txt | 20 +++++--------------
 1 file changed, 5 insertions(+), 15 deletions(-)

diff --git a/board/boundary/bootscripts/bootscript-yocto.txt b/board/boundary/bootscripts/bootscript-yocto.txt
index 5b098dacaa..4ea93fa849 100644
--- a/board/boundary/bootscripts/bootscript-yocto.txt
+++ b/board/boundary/bootscripts/bootscript-yocto.txt
@@ -68,15 +68,11 @@ if itest.s x${distro_bootpart} == x ; then
 	distro_bootpart=1
 fi
 
-if mmc dev 1; then
-    setenv devnum 1;
-    echo Using SD Card to load resinOS;
-fi;
-
 setenv resin_dev_index ${devnum};
 setenv resin_kernel_load_addr ${a_zImage}
 
 if env exists resin_set_kernel_root; then
+    setenv resin_uboot_device_types usb mmc;
     run resin_set_kernel_root;
     run set_os_cmdline;
 fi;
@@ -129,17 +125,11 @@ if itest.s "x" == "x${bpart}" ; then
 	bpart=2
 fi
 
-if test "sata" = "${devtype}" ; then
-	setenv bootargs "${bootargs} root=/dev/sda${bpart}" ;
-elif test "usb" = "${devtype}" ; then
-	setenv bootargs "${bootargs} root=/dev/sda${bpart}" ;
+if env exists resin_set_kernel_root; then
+    setenv bootargs "${bootargs} ${resin_kernel_root} ro rootwait ";
 else
-    if env exists resin_set_kernel_root; then
-        setenv bootargs "${bootargs} ${resin_kernel_root} ro rootwait ";
-    else
-        setenv bootargs "${bootargs} root=/dev/mmcblk${devnum}p2 ro rootwait ";
-    fi;
-fi
+    setenv bootargs "${bootargs} root=/dev/mmcblk${devnum}p2 ro rootwait ";
+fi;
 
 if itest.s "x" != "x${disable_msi}" ; then
 	setenv bootargs ${bootargs} pci=nomsi
-- 
2.17.1

